src/command_modules/azure-cli-network/azure/cli/command_modules/network/custom.py

```diff
@@ -175,30 +175,40 @@ def update_ag_frontend_port(instance, parent, item_name, port=None): # pylint: d
     return parent
 
 def create_ag_http_listener(resource_group_name, application_gateway_name, item_name,
-                            frontend_ip, frontend_port, ssl_cert=None, no_wait=False):
+                            frontend_ip, frontend_port, host_name=None, ssl_cert=None,
+                            no_wait=False):
     from azure.mgmt.network.models import ApplicationGatewayHttpListener
     ncf = _network_client_factory()
     ag = ncf.application_gateways.get(resource_group_name, application_gateway_name)
     new_listener = ApplicationGatewayHttpListener(
         name=item_name,
         frontend_ip_configuration=SubResource(frontend_ip),
         frontend_port=SubResource(frontend_port),
+        host_name=host_name,
+        require_server_name_indication=True if ssl_cert and host_name else None,
         protocol='https' if ssl_cert else 'http',
         ssl_certificate=SubResource(ssl_cert) if ssl_cert else None)
     _upsert(ag.http_listeners, new_listener, 'name', item_name)
     return ncf.application_gateways.create_or_update(
         resource_group_name, application_gateway_name, ag, raw=no_wait)
 
 def update_ag_http_listener(instance, parent, item_name, frontend_ip=None, frontend_port=None, # pylint: disable=unused-argument
-                            protocol=None, ssl_cert=None):
+                            host_name=None, ssl_cert=None):
     if frontend_ip is not None:
         instance.frontend_ip_configuration = SubResource(frontend_ip)
     if frontend_port is not None:
         instance.frontend_port = SubResource(frontend_port)
-    if protocol is not None:
-        instance.protocol = protocol
     if ssl_cert is not None:
-        instance.ssl_certificate = SubResource(ssl_cert)
+        if ssl_cert:
+            instance.ssl_certificate = SubResource(ssl_cert)
+            instance.protocol = 'Https'
+        else:
+            instance.ssl_certificate = None
+            instance.protocol = 'Http'
+    if host_name is not None:
+        instance.host_name = host_name if host_name else None
+    instance.require_server_name_indication = instance.host_name and \
+        instance.protocol.lower() == 'https'
```

Comment:
You can avoid the `.lower()` if you just make it lowercase above right?
Or does the service require that it be capitalized.